snowfall/objectives/ctc.py

```diff
@@ -0,0 +1,45 @@
+from typing import List, Tuple
+
+import torch
+from torch import nn
+
+import k2
+
+from snowfall.objectives.common import get_tot_objf_and_num_frames
+from snowfall.training.ctc_graph import CtcTrainingGraphCompiler
+
+
+class CTCLoss(nn.Module):
+    """
+    Connectionist Temporal Classification (CTC) loss.
+
+    TODO: more detailed description
+    """
+    def __init__(
+            self,
+            graph_compiler: CtcTrainingGraphCompiler,
+    ):
+        super().__init__()
+        self.graph_compiler = graph_compiler
+
+    def forward(
+            self,
+            nnet_output: torch.Tensor,
+            texts: List,
+            supervision_segments: torch.Tensor
+    ) -> Tuple[torch.Tensor, int, int]:
+        num = self.graph_compiler.compile(texts)
```

Comment:
Can we call the return value of `self.graph_compiler.compile(texts)` `num_graphs`
and the return value of `k2.intersect_dense(num, dense_fsa_vec, 10.0)` `num_lats`?